components/ILIAS/Mail/classes/Preview/class.ilStudyProgrammeMailTemplateContextPreview.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+class ilStudyProgrammeMailTemplateContextPreview extends ilStudyProgrammeMailTemplateContext
+{
+    public const ID = 'prg_context_manual_preview';
+
+    protected ilLanguage $lng;
+
+    public function getId(): string
+    {
+        return self::ID;
+    }
+
+    public function resolveSpecificPlaceholder(
+        string $placeholder_id,
+        array $context_parameters,
+        ilObjUser $recipient = null,
+        bool $html_markup = false
+    ): string {
+        switch ($placeholder_id) {
+            case ilStudyProgrammeMailTemplateContext::TITLE:
+                $string = 'programme title';
+                break;
```

Comment:
Could we add a new line between each case (after the `break;`) for a better readability?

And: Shouldn't this class be moved to the `StudyProgramme` component?

Reply:
I can do that.

As for the location: it could be either way as it's part of the StudyProgramme component but also part of the mail/preview component. If we move this we should also move "class.ilCourseMailTemplateTutorContextPreview.php".

What is your opinion here?

Reply:
We should move the course preview as well as the study programme preview.